src/Compilers/CSharp/Portable/Binder/Binder_Unsafe.cs

```diff
@@ -21,32 +22,33 @@ internal bool InUnsafeRegion
 
         /// <returns>True if a diagnostic was reported, or would have been reported if not for
         /// the suppress flag.</returns>
-        private bool ReportUnsafeIfNotAllowed(SyntaxNode node, DiagnosticBag diagnostics)
-        {
-            return ReportUnsafeIfNotAllowed(node, null, diagnostics);
-        }
-
-        /// <returns>True if a diagnostic was reported, or would have been reported if not for
-        /// the suppress flag.</returns>
-        private bool ReportUnsafeIfNotAllowed(SyntaxNode node, TypeSymbol sizeOfTypeOpt, DiagnosticBag diagnostics)
+        internal bool ReportUnsafeIfNotAllowed(SyntaxNode node, DiagnosticBag diagnostics, TypeSymbol sizeOfTypeOpt = null)
         {
             Debug.Assert((node.Kind() == SyntaxKind.SizeOfExpression) == ((object)sizeOfTypeOpt != null), "Should have a type for (only) sizeof expressions.");
-            return ReportUnsafeIfNotAllowed(node.Location, sizeOfTypeOpt, diagnostics);
+            return ReportUnsafeIfNotAllowed(node.Location, diagnostics, sizeOfTypeOpt);
         }
-
+        
         /// <returns>True if a diagnostic was reported, or would have been reported if not for
         /// the suppress flag.</returns>
-        internal bool ReportUnsafeIfNotAllowed(Location location, DiagnosticBag diagnostics)
+        internal bool ReportUnsafeIfNotAllowed(Location location, ref HashSet<DiagnosticInfo> useSiteDiagnostics, TypeSymbol sizeOfTypeOpt = null)
         {
-            return ReportUnsafeIfNotAllowed(location, sizeOfTypeOpt: null, diagnostics: diagnostics);
+            var diagnosticInfo = GetUnsafeDiagnosticInfo(sizeOfTypeOpt: sizeOfTypeOpt);
+
+            if (diagnosticInfo is null)
+            {
+                return false;
+            }
+
+            HashSetExtensions.InitializeAndAdd(ref useSiteDiagnostics, diagnosticInfo);
+            return true;
         }
 
         /// <returns>True if a diagnostic was reported, or would have been reported if not for
         /// the suppress flag.</returns>
```

Comment:
Good catch!

---
In reply to: [132313469](https://github.com/dotnet/roslyn/pull/21006#discussion_r132313469) [](ancestors = 132313469)